### PR TITLE
userspace: sample app domain applications

### PR DESCRIPTION
**Pull request by cmSAMIAMnot**
Sample using the app_memory submodule to protect memory
used by three threads.  This is a basic example tested on 96b_carbon,
frdm_k64f, and qemu_x86.  The main thread models an old
piece of hardware and should only be treated as a toy application.

Signed-off-by: Shawn Mosley <smmosle@tycho.nsa.gov>

### CODE REVIEW

File: samples/basic/userspace/shared_mem/README.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+to support more functionality as a toy, but the features are not completely
+implemented in the code.
+
+functionality
+*************
+The PT thread sends a message followed by the encrypted version of the
+message after sleeping.  To send the message the PT thread checks a
+flag, and if it is clear writes the message to a buffer shared with
+the encrypt thread.  After writing the buffer the flag is set. The encrypt
+thread copies the memory from the common buffer into the encrypted theads
```
Comment (dbkinder) [inline samples/basic/userspace/shared_mem/README.rst]: s/theads/thread's/

File: samples/basic/userspace/shared_mem/README.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+the actual enigma design.  This example is simply something interesting that
+seemed like a good use for the memory partitioning.  The software is designed
+to support more functionality as a toy, but the features are not completely
+implemented in the code.
+
+functionality
+*************
+The PT thread sends a message followed by the encrypted version of the
+message after sleeping.  To send the message the PT thread checks a
+flag, and if it is clear writes the message to a buffer shared with
```
Comment (dbkinder) [inline samples/basic/userspace/shared_mem/README.rst]: add a comma: ``clear, writes``

File: samples/basic/userspace/shared_mem/README.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+the encrypt thread.  After writing the buffer the flag is set. The encrypt
```
Comment (dbkinder) [inline samples/basic/userspace/shared_mem/README.rst]: add a comma:  ``buffer, the``

File: samples/basic/userspace/shared_mem/README.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+thread copies the memory from the common buffer into the encrypted theads
+private memory when the flag is set and then clears the flag.  Once the
+encrypted thread receives the text string it performs a simulation of
```
Comment (dbkinder) [inline samples/basic/userspace/shared_mem/README.rst]: add a comma: ``string, it``

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by cmSAMIAMnot** [comment]
I made the changes to README.rst and added a few sections.  The tests pass now, but I found a few hard coded numbers I am updating.  I will push again when I finish.

**Comment by cmSAMIAMnot** [comment]
This push should be ok.  I found a name conflict on the definition BLOCKSIZE for some boards.  I simply renamed my definition to fix the error.

**Review by AdithyaBaglody — DISMISSED** [review]
Missing license header.

**Review by dbkinder — APPROVED** [review]
doc change works for me, thanks.

**Comment by cmSAMIAMnot** [comment]
The last requested change was made three weeks ago based on comments from @AdithyaBaglody, and the license headers have been added.  Can this be merged now?

**Review by AdithyaBaglody — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**